### PR TITLE
Make ES index name configurable

### LINKED ISSUE #799: Make Elasticsearch index prefix configurable
If the index prefix was configurable people could deploy {collector, query} per k8s namespace and use one elasticsearch deployment.

Related to https://github.com/jaegertracing/jaeger-kubernetes/issues/85

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #799 

Dependency job PR:  https://github.com/jaegertracing/spark-dependencies/pull/38

### CODE REVIEW

**Comment by pavolloffay** [comment]
At the moment it does not add any delimiter after the prefix. We could add `-` or `:`.

**Comment by pavolloffay** [comment]
I have changed a couple of things.

* It 's backward compatible - if there is no index prefix it uses the same naming pattern `jaeger-<type>..`
* if the index prefix is specified it adds `:` at the end. e.g. prefix `foo` creates `foo:jaeger-<type>..`

**Comment by pavolloffay** [comment]
@yurishkuro @black-adder @vprithvi would you like to have a look?

File: plugin/storage/es/options.go [hunk 1]
```diff
@@ -141,6 +142,10 @@ func addFlags(flagSet *flag.FlagSet, nsConfig *namespaceConfig) {
 		nsConfig.namespace+suffixBulkFlushInterval,
 		nsConfig.BulkFlushInterval,
 		"A time.Duration after which bulk requests are committed, regardless of other tresholds. Set to zero to disable. By default, this is disabled.")
+	flagSet.String(
+		nsConfig.namespace+suffixIndexPrefix,
+		nsConfig.IndexPrefix,
+		"The prefix of Jaeger indices. For example \"production\" creates \"production:jaeger-*\". (Default unset)")
```
Comment (yurishkuro) [inline plugin/storage/es/options.go]: nit: Optional prefix for index names. (don't need "Default unset")

File: plugin/storage/es/dependencystore/storage.go [hunk 2]
```diff
@@ -107,18 +112,18 @@ func buildTSQuery(endTs time.Time, lookback time.Duration) elastic.Query {
...
-		currentIndex = indexName(ts)
+		currentIndex = indexName(prefix, ts)
 	}
 	return append(indices, firstIndex)
 }
 
-func indexName(date time.Time) string {
-	return dependencyIndexPrefix + date.UTC().Format("2006-01-02")
+func indexName(prefix string, date time.Time) string {
+	return prefix + dependencyIndex + date.UTC().Format("2006-01-02")
```
[the remark this answers is not in the document]
Comment (pavolloffay) [reply plugin/storage/es/dependencystore/storage.go]: It's per span write. Will update not sure if the cache would make a significant difference

File: plugin/storage/es/factory.go [hunk 3]
```diff
@@ -74,16 +74,16 @@ func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger)
 // CreateSpanReader implements storage.Factory
 func (f *Factory) CreateSpanReader() (spanstore.Reader, error) {
 	cfg := f.primaryConfig
-	return esSpanStore.NewSpanReader(f.primaryClient, f.logger, cfg.GetMaxSpanAge(), f.metricsFactory), nil
+	return esSpanStore.NewSpanReader(f.primaryClient, f.logger, cfg.GetMaxSpanAge(), f.metricsFactory, cfg.GetIndexPrefix()), nil
```
[the remark this answers is not in the document]
Comment (pavolloffay) [reply plugin/storage/es/factory.go]: I will keep the current form for the simplicity. If it grows we can refactor

**Pull request merged**